packages/mdc-textfield/README.md

```diff
@@ -329,102 +329,155 @@ Mixin | Description
-`trailingIconContent` | `string` (write-only) | Proxies to the foundation's `setTrailingIconContent` method.
-
-In addition to the above, the following properties proxy to the `input` element's properties of the same name:
-
-* `required`
-* `pattern`
-* `minLength`
-* `maxLength`
-* `min`
-* `max`
-* `step`
-
-Method Signature | Description
+Signature | Description
 --- | ---
+`emit(evtType: string, evtData: T, shouldBubble?: boolean) => void` | Fires a cross-browser-compatible custom event from the component root of the given type, with the given data.
 `focus() => void` | Focuses the `input` or `textarea` element.
+`initialSyncWithDOM() => void` | Initializes the Text Field's internal state based on the environment's state.
 `layout() => void` | Adjusts the dimensions and positions for all sub-elements.
+`listen(evtType: K, handler: SpecificEventListener<K>, options?: AddEventListenerOptions | boolean) => void` | Wrapper method to add an event listener to the component's root element. This is most useful when listening for custom events.
+`unlisten(evtType: K, handler: SpecificEventListener<K>, options?: AddEventListenerOptions | boolean) => void` | Wrapper method to remove an event listener to the component's root element. This is most useful when unlistening for custom events.
 
-## Usage Within Frameworks
+#### Properties
 
-If you are using a JavaScript framework, such as React or Angular, you can create a Text Field for your framework. Depending on your needs, you can use the _Simple Approach: Wrapping MDC Web Vanilla Components_, or the _Advanced Approach: Using Foundations and Adapters_. Please follow the instructions [here](../../docs/integrating-into-frameworks.md).
+Name | Type | Description
+--- | --- | ---
+`pattern` | `string` | @param Sets the input element's validation pattern.
+`disabled` | `boolean` | Proxies to the foundation's `isDisabled`/`setDisabled` methods.  @param Sets the Text Field disabled or enabled.
+`helperTextContent` | `string` | Sets the helper text element content. Proxies to the foundation's `setHelperTextContent` method when set.
+`leadingIconAriaLabel` | `string` | Sets the aria label of the leading icon. Proxies to the foundation's `setLeadingIconAriaLabel` method.
+`leadingIconContent` | `string` | Sets the text content of the leading icon. Proxies to the foundation's `setLeadingIconContent` method.
+`max` | `string` | @param Sets the input element's max.
+`maxLength` | `number` | @param Sets the input element's maxLength.
+`min` | `string` | @param Sets the input element's min.
+`minLength` | `number` | @param Sets the input element's minLength.
+`required` | `boolean` | @param Sets the Text Field to required.
+`ripple` | `MDCRipple | null` | The `MDCRipple` instance for the root element that `MDCTextField` initializes; this only applies to the default Text Field, and is `null` for other variants.
```

Comment:
I need to escape the `MDCRipple | null`...this isn't rendered the same on Github.